src/shared/utils/index.ts

```diff
@@ -74,38 +44,101 @@ export const rbtcMap = new Map([
   [undefined, false],
 ])
 
-export const formatTokenValues = (
-  number: string | number,
-  resultDecimals?: number,
-) => {
-  // make sure to use this only at the end when showing values
-  if (isNaN(Number(number))) {
-    return number.toString()
-  }
-  number = number.toString()
-  const asNumber = Number(number)
+interface FormatNumberOptions {
+  decimalPlaces?: number
+  useThousandSeparator?: boolean
+  isCurrency?: boolean
+  sign?: string
+}
 
-  if (asNumber < 1) {
-    return formatSmallNumbers(number, resultDecimals)
+/**
+ * Formats a number or a numeric string with the given options.
+ * @param num The number or string to format.
+ * @param options The formatting options to use.
+ * @returns The formatted number as a string.
+ */
+const formatNumber = (
+  num: number | string,
+  options: FormatNumberOptions = {},
+): string => {
+  // Attempt to parse if num is a string
+  if (typeof num === 'string') {
+    const parsedNum = parseFloat(num)
+    if (isNaN(parsedNum)) {
+      return num
+    }
+    num = parsedNum
   }
 
-  if (number.includes('.') && asNumber > 1) {
-    return formatWithDigits(number, resultDecimals)
+  // Destructure with default values
+  const {
+    decimalPlaces = 8,
+    useThousandSeparator = true,
+    isCurrency = false,
+    sign = '',
+  } = options
+
+  // Calculate the minimum value that can be displayed given the number of decimal places
+  const minValue = 1 / Math.pow(10, decimalPlaces)
+
+  // Check for small positive amounts less than the minimum value
+  if (num > 0 && num < minValue) {
+    return `<${sign}0.${'0'.repeat(decimalPlaces - 1)}1`
   }
 
-  if (number.length <= 3) {
-    return number
+  // Format the number with fixed decimal places
+  let result = num.toFixed(isCurrency ? 2 : decimalPlaces)
+
+  // For non-currency numbers, remove unnecessary trailing zeros
+  if (!isCurrency) {
+    result = result.replace(/(\.\d*?[1-9])0+$|\.0*$/, '$1')
   }
 
-  const longNumberArr = number.split('')
+  // Add thousand separators if enabled
+  if (useThousandSeparator) {
+    const parts = result.split('.')
+    parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ',')
+    result = parts.join('.')
+  }
 
-  for (let i = number.length - 3; i > 0; i -= 3) {
-    longNumberArr.splice(i, 0, ',')
+  // Check if the number is negative; if so, format accordingly
+  if (num < 0) {
+    return `-${sign}${result.substring(1)}`
   }
 
-  return longNumberArr.join('')
+  return `${sign}${result}`
 }
 
+/**
+ * Formats a number or a numeric string as a USD value with a dollar sign.
+ * Should be used only at the end when showing values.
+ * @param value The number or string to format.
+ * @returns The formatted USD value with a dollar sign as a string.
+ */
+export const formatUsdValue = (value: number | string): string =>
```

Comment:
let's rename that to `formatFiatValue` to prepare for the future? We can change the values in components in the future PRs